pkg/cmd/util/clientcmd/factory.go

```diff
@@ -281,7 +281,14 @@ func (c *clientCache) ClientConfigForVersion(version string) (*kclient.Config, e
 		if err != nil {
 			return nil, err
 		}
-		// TODO: We want to reuse the ustream negotiation logic, which is coupled
+		c.defaultConfig = config
+	}
+	// TODO: have a better config copy method
+	if config, ok := c.configs[version]; ok {
+		return config, nil
+	}
+	if c.negotiatingClient == nil {
```

Comment:
building the negotiating client is logically separate from building the default config, and we should only do it after we're sure we need to negotiate (having checked our config cache)